File: web/src/api/emailTemplateFolder.js
import service from '@/utils/request'

export const getEmailTemplateFolderTree = () => {
  return service({
    url: '/emailTemplateFolder/tree',
    method: 'get'
  })
}

export const createEmailTemplateFolder = (data) => {
  return service({
    url: '/emailTemplateFolder/create',
    method: 'post',
    data
  })
}

export const updateEmailTemplateFolder = (data) => {
  return service({
    url: '/emailTemplateFolder/update',
    method: 'put',
    data
  })
}

export const deleteEmailTemplateFolder = (data) => {
  return service({
    url: '/emailTemplateFolder/delete',
    method: 'delete',
    data
  })
}

File: server/router/system/email_template_folder.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type EmailTemplateFolderRouter struct{}

func (r *EmailTemplateFolderRouter) InitEmailTemplateFolderRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	group := Router.Group("emailTemplateFolder").Use(middleware.OperationRecord())
	groupWithoutRecord := Router.Group("emailTemplateFolder")
	{
		group.POST("create", emailTemplateFolderApi.CreateEmailTemplateFolder)
		group.PUT("update", emailTemplateFolderApi.UpdateEmailTemplateFolder)
		group.DELETE("delete", emailTemplateFolderApi.DeleteEmailTemplateFolder)
	}
	{
		groupWithoutRecord.GET("tree", emailTemplateFolderApi.GetEmailTemplateFolderTree)
	}
}

